src/components/Dashboard/Chart/Chart.js

```diff
@@ -80,6 +88,48 @@ function getVestingData(start, cliff, duration, total, revokeLog) {
   )
 }
 
+function getVestingDataV2(
+  start,
+  cliff,
+  duration,
+  periodDuration,
+  vestedPerPeriod,
+  linear
+) {
+  const cliffEndDay = getCliffEndDay(start, cliff)
+  const vestingDays = getDurationInDays(duration)
+
+  let vestingData = []
+
+  // TODO: This can probably be optimized to avoid doing the same calculations for every day.
+  for (let i = 0; i < vestingDays; i++) {
+    let vestedThatDay
+
+    if (i <= cliffEndDay) {
+      vestedThatDay = 0
+    } else {
+      const elapsedPeriods = (DAY_IN_SECONDS * (i + 1)) / periodDuration
```

Comment:
we could add a check for `periodDuration > 0`

Reply:
Addressed in https://github.com/decentraland/vesting-dashboard/pull/45/commits/f9d176b781f7d5719c403352968a83cd1443fe52